lib/Doctrine/DBAL/Query/Expression/CompositeExpression.php

```diff
@@ -43,47 +47,48 @@ class CompositeExpression implements Countable
      */
     public function __construct(string $type, array $parts = [])
     {
-        $this->type = $type;
-
-        $this->addMultiple($parts);
+        $this->type  = $type;
+        $this->parts = array_values(array_filter($parts, static function ($part) {
+            return ! ($part instanceof self && count($part) === 0);
```

Comment:
We'll be able to remove these checks/filters once we'll make sure that `CompositeExpression` always contains at least one part. I'll make another PR to fix #3845 and update this code later.

Reply:
Why is this filter needed now?

Reply:
Because it accepts empty `CompositeExpression` objects, that were filtered so far by `addMultiple()`.